### ISSUE TITLE
limit ValueError: The argument 'nan' is not comparable.

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#16611]
I got this exception when calculate `limit((cos((k + 1)/2) - 1)/(cos(k/2) - 1)*x, k, oo)`. Possibly, should give some interval.

**Comment by jksuom** [comment]
The coefficient of `x` is nonnegative. The result should probably be `AccumBounds(0, oo)*x` (or `nan`). It seems that the computation of the product `AccumBounds(-2, 0)*AccumBound(-oo, -S(1)/2)` (limits of  `cos((k + 1)/2) - 1` and `1/(cos(k/2) - 1)`) fails with ValueError.

**Comment by SidDsloth** [comment]
What is x in the expression when the limit is being found wrt k?

**Comment by proy87** [comment]
x = Symbol('x')

**Comment by SidDsloth** [comment]
I did comprehend that it is a symbol however failed to understand its significance in the expression.

**Comment by smichr** [comment]
The root cause is fixed in #21523:
```python
>>> limit((cos((k + 1)/2) - 1)/(cos(k/2) - 1)*x, k, oo)
AccumBounds(-oo, oo)*x
```
